### ISSUE TITLE
ManyToMany relation wrongly validates values_list field name

### ISSUE THREAD

**Issue by Andrioden** [typeddjango/django-stubs#2284]
# Bug report

The following code gives error when it should not

``` python
class Cube(Model):
    heat = IntegerField(default=0)


class Ice(Model):
    cubes = ManyToManyField(Cube, blank=True)

    def heat_sum(self) -> int:
        return sum(self.cubes.values_list("heat", flat=True)) # <-- Error
```

## What's wrong

mypy gives error

`error: Cannot resolve keyword 'heat' into field. Choices are: cube, cube_id, ice, ice_id, id  [misc]`

## How is that should be

No error

## System information

- OS: windows
- `python` version:  3.12.1
- `django` version: 5.0.3
- `mypy` version: 1.11.0
- `django-stubs` version: 5.0.3
- `django-stubs-ext` version: 5.0.3

**Comment by flaeppe** [comment]
Thank you for the report. There's a fix in #2283 that hasn't been merged yet

**Comment by ruudvandenboomen** [comment]
I'm still facing this issue, doesn't seem to be fixed on version 5.0.4 in combination with Django 5.1

**Comment by Andrioden** [comment]
my problem was solved with 5.0.4 at least
